File: external/hypergraph-of-entity/src/main/java/armyant/hypergraphofentity/nodes/DocumentNode.java
package armyant.hypergraphofentity.nodes;

/**
 * Created by jldevezas on 2017-10-24.
 */
public class DocumentNode extends Node {
    public DocumentNode() {
    }

    public DocumentNode(String name) {
        super(name);
    }

    @Override
    public String getType() {
        return "document";
    }
}

File: external/cpp-impl/src/hgoe/nodes/document_node.cpp
//
// Created by jldevezas on 3/8/18.
//

#include <hgoe/nodes/document_node.h>

BOOST_CLASS_EXPORT_IMPLEMENT(DocumentNode)

DocumentNode::DocumentNode() = default;

DocumentNode::DocumentNode(std::string name) : Node(boost::move(name)) {}

Node::Label DocumentNode::label() const {
    return Label::DOCUMENT;
}

void DocumentNode::print(std::ostream &os) const {
    os << "DocumentNode ";
    Node::print(os);
}
